pkg/resolution/resolver/git/resolver_test.go

```diff
@@ -212,240 +247,241 @@ func TestResolve(t *testing.T) {
+		expectedStatus: createFailureStatus(),
+		expectedErr:    createError("cannot get API token, required when specifying 'repo' param, 'api-token-secret-name' not specified in config"),
+	}, {
+		name: "api: token secret key not specified",
+		args: &params{
 			revision:   "main",
 			pathInRepo: "pipelines/example-pipeline.yaml",
 			org:        testOrg,
 			repo:       testRepo,
-			config: map[string]string{
-				ServerURLKey:          "fake",
-				SCMTypeKey:            "fake",
-				APISecretNameKey:      "token-secret",
-				APISecretNamespaceKey: system.Namespace(),
-			},
-			apiToken:       "some-token",
-			expectedStatus: createFailureStatus(),
-			expectedErr:    createError("cannot get API token, required when specifying 'repo' param, 'api-token-secret-key' not specified in config"),
-		}, {
-			name:       "api: SCM type not specified",
+		},
+		config: map[string]string{
+			ServerURLKey:          "fake",
+			SCMTypeKey:            "fake",
+			APISecretNameKey:      "token-secret",
+			APISecretNamespaceKey: system.Namespace(),
+		},
+		apiToken:       "some-token",
+		expectedStatus: createFailureStatus(),
+		expectedErr:    createError("cannot get API token, required when specifying 'repo' param, 'api-token-secret-key' not specified in config"),
+	}, {
+		name: "api: SCM type not specified",
+		args: &params{
 			revision:   "main",
 			pathInRepo: "pipelines/example-pipeline.yaml",
 			org:        testOrg,
 			repo:       testRepo,
-			config: map[string]string{
-				APISecretNameKey:      "token-secret",
-				APISecretKeyKey:       "token",
-				APISecretNamespaceKey: system.Namespace(),
-			},
-			apiToken:       "some-token",
-			expectedStatus: createFailureStatus(),
-			expectedErr:    createError("missing or empty scm-type value in configmap"),
 		},
+		config: map[string]string{
+			APISecretNameKey:      "token-secret",
+			APISecretKeyKey:       "token",
+			APISecretNamespaceKey: system.Namespace(),
+		},
+		apiToken:       "some-token",
+		expectedStatus: createFailureStatus(),
+		expectedErr:    createError("missing or empty scm-type value in configmap"),
+	},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			ctx, _ := ttesting.SetupFakeContext(t)
 
+			// for scm API git testing
 			fakeClone := fmt.Sprintf("https://fake/%s/%s.git", testOrg, testRepo)
 			resolver := &Resolver{
 				clientFunc: func(driver string, serverURL string, token string, opts ...factory.ClientOptionFunc) (*scm.Client, error) {
 					scmClient, scmData := fake.NewDefault()
 
-					// Currently, we only could care about repositories.
+					// repository service
 					scmData.Repositories = []*scm.Repository{{
 						FullName: fmt.Sprintf("%s/%s", testOrg, testRepo),
 						Clone:    fakeClone,
 					}}
 
+					// git service
+					scmData.Commits = map[string]*scm.Commit{
+						"main":  {Sha: revisionToSHA["main"]},
+						"other": {Sha: revisionToSHA["other"]},
+					}
 					return scmClient, nil
 				},
 			}
```

Comment:
nit: This data all looks like it doesn't change. If you don't need a unique resolver for each test case, consider hoisting this out of the test loop.

Reply:
👍  SGTM!!!